src/aws_encryption_sdk/materials_managers/default.py

```diff
@@ -101,6 +107,64 @@ def get_encryption_materials(self, request):
             signing_key=signing_key,
         )
 
+    def _get_encryption_materials_using_keyring(self, request):
+        """Creates encryption materials using underlying keyring.
+
+        :param request: encryption materials request
+        :type request: aws_encryption_sdk.materials_managers.EncryptionMaterialsRequest
+        :returns: encryption materials
+        :rtype: aws_encryption_sdk.materials_managers.EncryptionMaterials
+        :raises InvalidCryptographicMaterialsError: if keyring cannot complete encryption materials
+        :raises InvalidCryptographicMaterialsError:
+           if encryption materials received from keyring do not match request
+        """
+        algorithm = request.algorithm if request.algorithm is not None else self.algorithm
+        encryption_context = request.encryption_context.copy()
+
+        signing_key = self._generate_signing_key_and_update_encryption_context(algorithm, encryption_context)
+
+        expected_encryption_context = encryption_context.copy()
+
+        encryption_materials = EncryptionMaterials(
+            algorithm=algorithm, encryption_context=encryption_context, signing_key=signing_key,
+        )
+
+        final_materials = self.keyring.on_encrypt(encryption_materials=encryption_materials)
+
+        if not final_materials.is_complete:
+            raise InvalidCryptographicMaterialsError("Encryption materials are incomplete!")
+
+        materials_are_valid = (
```

Comment:
Who validates that you have a plaintext data key?
And that it matches the suite that you have?
Is that in the materials?

Reply:
The materials do. That's what we're checking on line 134 where we ask the materials if they're complete.

https://github.com/aws/aws-encryption-sdk-python/blob/952bce44db630754de6a0dad451f72665f83b1c7/src/aws_encryption_sdk/materials_managers/__init__.py#L248-L264